internal/services/network/network_manager_deployment_resource.go

```diff
@@ -346,13 +347,21 @@ func resourceManagerDeploymentWaitForFinished(ctx context.Context, client *netwo
 		Delay:      10 * time.Second,
 		Pending:    []string{"NotStarted", "Deploying"},
 		Target:     []string{"Deployed"},
-		Refresh:    resourceManagerDeploymentResultRefreshFunc(ctx, client, managerDeploymentId),
-		Timeout:    d,
+		Refresh: func() (interface{}, string, error) {
+			result, state, err := resourceManagerDeploymentResultRefreshFunc(ctx, client, managerDeploymentId)()
+			if state == "NotFound" {
```

Comment:
is there a reason to wrap this function instead of adding this to that function?

Reply:
The refresh function is used by another [resourceManagerDeploymentWaitForDeleted function](https://github.com/hashicorp/terraform-provider-azurerm/pull/24330/files#diff-e87548b6c4289d26dc6462ab7e28eda62e4717f567de763344b60ad1547af678L325) to wait until the resource is surely deleted, so we cannot adding this to that function (`resourceManagerDeploymentWaitForDeleted` does not need certain times of NotFoundCheck, but only `resourceManagerDeploymentWaitForFinished` need the NotFoundCheck).

Reply:
The [NotFoundCheck ](https://github.com/hashicorp/terraform-provider-azurerm/blob/dca110857840aee7fdc035d9085fcf778aa16d7f/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry/state.go#L136-L147)is to error out if it is not found for several times.

Reply:
If this is different and unique from other endpoints this sounds like a potential bug?

could we please open a bug and link to it / detail why this is different/special compared to the other uses of the funciton?

Reply:
submit API issue for this https://github.com/Azure/azure-rest-api-specs/issues/27327